### DISCUSSION TITLE
Is there a way to use composite primary keys with entities exteneded from BaseEntity?

### DISCUSSION THREAD

**Discussion by yf-hk** [mikro-orm/mikro-orm#2298] [Q&A]
```ts
export declare abstract class BaseEntity<T extends AnyEntity<T>, PK extends keyof T, P extends Populate<T> | unknown = unknown> implements IWrappedEntity<T, PK, P> {
    constructor();
    isInitialized(): boolean;
    populated(populated?: boolean): void;
    toReference(): any;
    toObject(ignoreFields?: string[]): Dictionary;
    toJSON(...args: any[]): Dictionary;
    toPOJO(): EntityData<T>;
    assign(data: EntityData<T>, options?: AssignOptions): T;
    init(populated?: boolean): Promise<T>;
}
```
From the definition, it seems to be impossible. What about changing it to `PK extends (Primary<T> | Primary<T>[])`, which is similar to `em.getReference`'s argument?

**Comment by nurdism** [comment]
is there any solution for this? Is this needed for composite primary keys?

**Reply by yf-hk** [reply]
Not as I know. I ended up not using BaseEntity and wrap the entity manually

**Reply by B4nan** [reply]
I have no idea what were actually asking here. Of course you "can use composite PKs with base entity". If you want answers, you need to ask less cryptic questions :P

**Reply by yf-hk** [reply]
From the TypeScript definition, the second generic of BaseEntity is PK, which is `keyof T`, which makes it impossible to use an array as the PK. That's what I was asking. Is it possible to define an entity extending BaseEntity with the correct composite PKs? Or just pass `any` to PK, will there be any problems?

**Reply by B4nan** [reply]
> which is keyof T, which makes it impossible to use an array as the PK

Well, but you can use union, it shouldn't be array. 

```ts
@Entity()
export class User extends BaseEntity<User, 'firstName' | 'lastName'> {

  @PrimaryKey()
  firstName!: string;
  
  @PrimaryKey()
  lastName!: string;

}
```

**Reply by yf-hk** [reply]
I see, thanks for your help! I'll try it and see if it works
